### PR TITLE
Seasonal finetuning pipeline

### PR DESCRIPTION
**Pull request by cbutsko**
Add end-to-end seasonal finetuning pipeline with dual-task (LC + CT) training, annual season support, and quality-based sample weighting.

**Seasonality data:**
- Add tc-annual to SEASONALITY_COLUMN_MAP
- Update seasonality_lookup.parquet with annual season columns
- Add regenerate_seasonality_lookup.py utility

**Training core (src/worldcereal/train/):**
- datasets.py: season mask resolution with nodata DOY fallback, SensorMaskingConfig, DualHeadBatchSampler, configurable season coverage thresholds, annual season support
- data.py: remove_small_classes returns removed classes, region filtering, dual-task label annotation helpers
- finetuning_utils.py: SeasonalMultiTaskLoss with per-task sample weighting, val loss EMA smoothing, relaxed early stopping
- Set MIN_EDGE_BUFFER = 2

**Finetuning script:**
- finetune_presto.py: dual-task annotation, multiplicative quality x outlier sample weighting, zero-quality filtering, map-sampled LC quality hotfix, temporal validity pre-check, effective classes, head manifest, full CLI
- SLURM and local shell scripts for training

**Tests:**
- Add test_season_coverage.py for season coverage logic

### CODE REVIEW

File: scripts/training/finetuning/finetune_presto.py [hunk 1]
```diff
@@ -1169,10 +2046,91 @@ def auto_or_int(value):
...
+
+    # Season coverage threshold for the validation/test splits.
+    parser.add_argument(
+        "--eval_min_season_coverage",
+        type=float,
+        default=1.0,
+        help=(
+            "Minimum fraction of a season's composite slots required for val/test "
+            "splits.  When omitted, uses the default value of 1.0 (which can cause "
+            "issues for annual seasons while calendars allow for annual seasons to be >12 months)."
```
Comment (kvantricht) [inline scripts/training/finetuning/finetune_presto.py]: annual seasons should (no longer/)not be > 12 months?

File: scripts/training/finetuning/finetune_presto.py [hunk 2]
```diff
@@ -1169,10 +2046,91 @@ def auto_or_int(value):
...
+
+    # Season selection
+    parser.add_argument(
+        "--season_ids",
+        type=str,
+        nargs="*",
+        default=None,
+        help=(
+            "Season IDs for crop-type supervision (e.g. 'tc-s1 tc-s2' or 'tc-annual'). "
+            "Defaults to GLOBAL_SEASON_IDS (tc-s1, tc-s2). "
```
Comment (kvantricht) [inline scripts/training/finetuning/finetune_presto.py]: don't see that default here?

**Review by kvantricht — APPROVED** [review]
